addons/point_of_sale/models/account_journal.py

```diff
@@ -14,3 +14,10 @@ def _check_type(self):
         methods = self.env['pos.payment.method'].sudo().search([("journal_id", "in", self.ids)])
         if methods:
             raise ValidationError(_("This journal is associated with a payment method. You cannot modify its type"))
+
+    def _get_journal_inbound_outstanding_payment_accounts(self):
+        res = super()._get_journal_inbound_outstanding_payment_accounts()
+        account_ids = set(res.ids)
+        for payment_method in self.pos_payment_method_ids:
+            account_ids.add(payment_method.outstanding_account_id.id or self.company_id.account_journal_payment_debit_account_id.id)
```

Comment:
Maybe you can wrap this inside a try catch that expects access error. If so, no need to attempt adding anything to the account_ids. Note that you should make sure that other types of errors should blow up the stack, only access error should be ignored. I'm reviewing because @rhe-odoo told me about this.